selectors/selectors.md

```diff
@@ -108,3 +108,271 @@ These components imply or expand into the following things:
+
+```json
+{
+  "show": "start-trek-voyager",
+  "characters": {
+    "kathrin-janeway": {
+      "birthday": {
+        "day": 20,
+        "month": 5,
+        "year": 2328
+      },
+      "rank": "vice admiral"
+    }
+  }
+}
+```
+
+The final Selector could look like this:
+
+```json
+{
+  "cidRootedSelector": {
+    "root": "cidabcdef",
+    "selectors": [
+      {"selectPath": "characters"},
+      {"selectPath": "kathryn-janeway"},
+      {"selectPath": "birthday"},
+      {"selectPath": "year"}
+    ]
+  }
+}
+```
+
+
+### Getting a certain number of parent blocks in a blockchain
+
+You want to get a certain number of parents from a certain block.
+
+The shape of a block could look like this (in JSON):
+
+```json
+{
+  "parent": "parentcid",
+  "time": 1549641260,
+  "none": 3423545
+}
+```
+
+If you know you want five parents you could use Path Selectors:
+
+```json
+{
+  "cidRootedSelector": {
+    "root": "cidabcdef",
+    "selectors": [
+      {"selectPath": "parent"},
+      {"selectPath": "parent"},
+      {"selectPath": "parent"},
+      {"selectPath": "parent"},
+      {"selectPath": "parent"}
+    ]
+  }
+}
+```
+
+But there's also a better way with using a Recursive Selector with a limit:
+
+```json
+{
+  "cidRootedSelector": {
+    "root": "cidabcdef",
+    "selectors": [
+      {"selectRecursive": {
+        "follow": [
+          {"selectPath": "parent"}
+        ],
+        "depthLimit": 5
+      }}
+    ]
+  }
+}
+```
+
+
+### Getting changes up to a certain one
+
+This use case is inspired by CRDTs, where you have a chain of changes. You observe a new change and want to get all the previous changes up to the one that you have already observed. It is a recursive query with a CID as stopping condition.
+
+The shape of a change could look like this (in JSON):
+
+```json
+{
+  "prev": "prevcid",
+  "timestamp": 1549641260,
+  "value": "abc"
+}
+```
+
+It will be a Recursive Selector following a long a certain field (`prev` in this case):
+
+```json
+{
+  "cidRootedSelector": {
+    "root": "cidabcdef",
+    "selectors": [
+      {"selectCid": "cidabcdef"},
+      {"selectRecursive": {
+        "follow": [
+          {"selectPath": "prev"}
+        ],
+        "cidLimit": "somecid"
+      }}
+    ]
+  }
+}
+```
+
+
+### Getting a full sub-DAG
+
+For getting a full file from UnixFSv1 you need to retrieve a full sub-DAG.
+
+An example selector to get the full sub-DAG rooted at a certain CID:
+
+
+```json
+{
+  "cidRootedSelector": {
+    "root": "cidabcdef",
+    "selectors": [
+      {"selectRecursive": {
+        "follow": [
+          {"selectPath": "Links"},
+          {"selectArrayAll": null},
+          {"selectPath": "multihash"}
+        ]
+      }}
+    ]
+  }
+}
+```
+
+If it's a file in some directory, you can also start at a deeper level:
+
+```json
+{
+  "cidRootedSelector": {
+    "root": "cidabcdef",
+    "selectors": [
+      {"selectCid": "cidabcdef"},
```

Comment:
same question as above, is this serving an important purpose or demonstrating the possible redundancy?